File: app/src/main/java/com/aquaguard/data/local/dao/UsageDao.java
package com.aquaguard.data.local.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.aquaguard.data.local.entity.UsageEntity;

import java.util.List;

@Dao
public interface UsageDao {
    @Query("SELECT * FROM water_usage WHERE deviceId = :deviceId ORDER BY date DESC")
    LiveData<List<UsageEntity>> getUsageHistory(String deviceId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertUsage(UsageEntity usage);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<UsageEntity> usages);
}

File: Aqua Guard/frontend/app/src/main/java/com/aquaguard/data/local/dao/UsageDao.kt
package com.aquaguard.data.local.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.aquaguard.data.local.entity.UsageEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface UsageDao {
    @Query("SELECT * FROM water_usage WHERE deviceId = :deviceId ORDER BY date DESC")
    fun getUsageHistory(deviceId: String): Flow<List<UsageEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertUsage(usage: UsageEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(usages: List<UsageEntity>)
}
